src/main/java/org/openstreetmap/atlas/checks/validation/tag/GenericTagCheck.java

```diff
@@ -0,0 +1,939 @@
+     *
+     * @param instructions
+     *            The instruction collection to append instructions to
+     * @param tag
+     *            The specific tag to check
+     * @param checkInfo
+     *            Wiki Data information (specific, if available)
+     * @return {@code true} if the regex failed
+     */
+    private boolean checkRegex(final Map<String, Collection<IFeatureChange>> instructions,
+            final Map.Entry<String, String> tag, final WikiData checkInfo)
+    {
+        // "P13" is "value validation regex". We must wrap with "^(" and ")$"
+        if (checkInfo != null && checkInfo.getValueValidationRegexP13() != null)
+        {
+            // Reuse the compiled patterns
+            final var pattern = checkInfo.getValueValidationRegexP13();
+            if (!pattern.matcher(tag.getValue()).matches())
+            {
+                instructions.put(this.getLocalizedInstruction(
+                        FALLBACK_INSTRUCTIONS.indexOf(INSTRUCTION_MISMATCHED_REGEX), tag.getValue(),
+                        pattern.toString(), tag.getKey(), checkInfo.getId()),
+                        Collections.emptyList());
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Do relation specific checks
+     *
+     * @param instructions
+     *            Instructions to add to
+     * @param The
+     *            object to check
+     * @param checkInfo
+     *            The wiki information for the tag to be checked
+     * @param tag
+     *            The tag to be checked
+     * @return {@code true} if the relation had a problem
+     */
+    private boolean checkRelations(final Map<String, Collection<IFeatureChange>> instructions,
+            final Taggable object, final Map.Entry<String, String> tag, final WikiData checkInfo)
+    {
+        if (checkInfo != null && object instanceof Relation && checkInfo.isUseOnRelationsP36()
+                && "type".equals(tag.getKey()))
+        {
+            final var relation = (Relation) object;
+            final var relationInfo = this.getWikiData(
+                    Map.of(WikiProperty.PERMANENT_RELATION_TYPE_ID_P41.getId(), tag.getValue()));
+            if (relationInfo == null)
+            {
+                instructions.put(
+                        this.getLocalizedInstruction(
+                                FALLBACK_INSTRUCTIONS.indexOf(INSTRUCTION_UNKNOWN_RELATION_TYPE),
+                                tag.getValue(), relation.getOsmIdentifier(), checkInfo.getId()),
+                        Collections.emptyList());
+                return true;
+            }
+            final Collection<String> additionalInstructions = new TreeSet<>();
+            for (final RelationMember relationMember : relation.members())
+            {
+                // Empty roles should exist in WikiData
+                final String role = relationMember.getRole();
+                final WikiData roleInfo = this
```

Comment:
One more to convert to var
```suggestion
                final var roleInfo = this
```

Reply:
It is always fun when you miss _one_.